src/katsdpcontroller/generator.py

```diff
@@ -1146,10 +1147,11 @@ def get_baseline_index(a1, a2):
             sum(stream.data_rate() for stream in streams) / n_engines
         )
         xbgpu.gpus = [scheduler.GPURequest()]
-        xbgpu.gpus[0].compute = (
-            0.15 * bw_scale
-        )  # TODO: NGC-1222 update depending on number and type of streams
         xbgpu.gpus[0].mem = 300 + _mb(3 * chunk_size)
+        # This memory is independent of the number of beams, but is needed
+        # only if there is at least one beam. So set it to 0 here then
+        # update it if we see a beam.
+        rand_state_size = 0
```

Comment:
Is this currently true? Or does it depend on the update in https://github.com/ska-sa/katgpucbf/pull/731?

Either way, noted.